src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoDetailsStep.tsx

```diff
@@ -24,27 +29,51 @@ const StackRepoDetailsStep: React.FC<StackWizardStepProps> = (props): JSX.Elemen
         initialValues={{
           repo: '',
           branch: '',
+          gitFetchType: GitFetchTypes.Branch,
           scriptsPath: '',
           ...prevStepData
         }}
         formName={`resourcestack-wizard-${identifier}`}
         onSubmit={formData => nextStep?.({ ...prevStepData, ...formData })}
         validationSchema={Yup.object({
-          repo: Yup.string().required(getString('common.git.validation.repoRequired')),
-          branch: Yup.string().required(getString('common.git.validation.branchRequired')),
-          scriptsPath: Yup.string().required(getString('iacm.stackWizard.scriptsPathRequired'))
+          repo: Yup.string().when('repoConnector.scope', {
+            is: value => value === 'account',
+            then: Yup.string().required(getString('common.git.validation.repoRequired')),
+            otherwise: Yup.string().notRequired()
+          }),
+          scriptsPath: Yup.string().required(getString('iacm.stackWizard.scriptsPathRequired')),
+          gitFetchType: Yup.string().required(getString('validation.thisIsARequiredField')),
+          branch: Yup.string().when('gitFetchType', {
+            is: value => value === GitFetchTypes.Branch,
+            then: Yup.string().required(getString('common.git.validation.branchRequired')),
+            otherwise: Yup.string().required(getString('validation.commitId'))
+          })
         })}
         enableReinitialize
       >
         {formik => {
-          const { isValid } = formik
+          const { isValid, values } = formik
+          const isBranchFetchType = values.gitFetchType === GitFetchTypes.Branch
           return (
             <FormikForm>
               <Layout.Vertical flex={{ justifyContent: 'space-between', alignItems: 'flex-start' }}>
                 <Layout.Vertical width={400}>
-                  <FormInput.MultiTextInput name="repo" label={getString('repository')} />
-                  <FormInput.MultiTextInput name="branch" label={getString('gitBranch')} />
-                  <FormInput.MultiTextInput name="scriptsPath" label={getString('iacm.stackWizard.scriptsPath')} />
+                  {values?.repoConnectorObject?.scope === 'account' && (
+                    <FormInput.Text name="repo" label={getString('repository')} />
+                  )}
+                  <FormInput.Select
+                    name="gitFetchType"
+                    label={getString('pipeline.manifestType.gitFetchTypeLabel')}
+                    items={[
+                      { label: getString('gitFetchTypes.fromBranch'), value: 'branch' },
+                      { label: getString('gitFetchTypes.fromCommit'), value: 'commitId' }
```

Comment:
Better to use your enum here to ensure consistency
```suggestion
                      { label: getString('gitFetchTypes.fromBranch'), value: GitFetchTypes.Branch },
                      { label: getString('gitFetchTypes.fromCommit'), value: GitFetchTypes.Commit }
```